src/nav_msg_glo.c

```diff
@@ -0,0 +1,285 @@
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+#include <libswiftnav/nav_msg_glo.h>
+#include <libswiftnav/time.h>
+
+/* local containers for parsing GLO ephemeris*/
+static ephemeris_t g_e;
+static bool done_flag = false;
+static u16 nt;
+static u8 n4;
+static u8 hrs;
+static u8 min;
+static u8 sec;
+
+/* Word Ft (accuracy of measurements), refer to GLO ICD, Table 4.4 */
+const float f_t[] = {1.0f,
+                     2.0f,
+                     2.5f,
+                     4.0f,
+                     5.0f,
+                     7.0f,
+                     10.0f,
+                     12.0f,
+                     14.0f,
+                     16.0f,
+                     32.0f,
+                     64.0f,
+                     128.0f,
+                     256.0f,
+                     512.0f,
+                     -1.0f};
+
+/* Word P1 (Time interval between adjacent values of tb, minutes), refer Table 4.3 */
+const u8 p1[] = {0, 30, 45, 60}; /* min */
+
+void nav_msg_init_glo(nav_msg_glo_t *n)
+{
+  /* Initialize the necessary parts of the nav message state structure. */
+  memset(n, 0, sizeof(nav_msg_glo_t));
+  n->next_string_id = 1; /* start parsing from string 1 */
+  n->state = SYNC_TM;
+  done_flag = false;
+}
+
+/** Extract a word of n_bits length (n_bits <= 32) at position bit_index into
+ * the subframe. Refer to bit index to Table 4.6 and 4.11 in GLO ICD 5.1 (pg. 34)
+ * \param n pointer to GLO nav message structure to be parsed
+ * \param bit_index number of bit the extract process start with. Range [1..85]
+ * \param n_bits how many bits should be extracted [1..32]
+ * \return word extracted from navigation string
+ */
+/*static*/ u32 extract_word_glo(const nav_msg_glo_t *n, u16 bit_index, u8 n_bits)
+{
+  assert(n_bits <= 32 && n_bits > 0);
+  assert(bit_index <= 85 && bit_index > 0);
+
+  /* Extract a word of n_bits length (n_bits <= 32) at position bit_index into
+   * the GLO string.*/
+  bit_index--;
+  u32 word = 0;
+  u8 bix_hi = bit_index >> 5;
+  u8 bix_lo = bit_index & 0x1F;
+  if (bix_lo + n_bits <= 32) {
+    word = n->string_bits[bix_hi] >> bix_lo;
+    word &= (0xffffffff << (32-n_bits)) >> (32-n_bits);
+  } else {
+    u8 s = 32-bix_lo;
+    word = extract_word_glo(n, bit_index+1, s) |
+           extract_word_glo(n, bit_index+1 + s, n_bits - s) << s;
+  }
+
+  return word;
+}
+
+/** Navigation message GLO decoding update.
+ * Called once per nav bit interval (10 ms).
+ * Performs the necessary steps to store the nav bits in buffer.
+ *
+ * \param n GLO Nav message decode state struct
+ * \param bit_val State of the nav bit to process, 0 or 1
+ *
+ * \return  1 if Glo nav string ready for decoding,
+ *         -1 otherwise.
+ */
+s8 nav_msg_update_glo(nav_msg_glo_t *n, bool bit_val)
+{
+  s8 ret = -1;
+
+  switch (n->state) {
```

Comment:
switch-case formatting does not match SN C coding style guide: https://www.kernel.org/doc/Documentation/CodingStyle:
should be:
switch () {
case x:
  ..
}